Begin working Morello into the exercise documentation. (#36)
CTSRD-CHERI/cheri-exercises: src/introduction/README.md

This set of exercises and adversarial missions is intended to:

- Build a baseline skill set with unmodified instruction set
  architectures (ISAs) such as ARMv8 and RISC-V and the same
  baseline ISAs with CHERI extensions, as well as awareness
  of some of the dynamics of CHERI-enabled software, through skills development
  exercises.

> - Develop adversarial experience with CHERI enabled ISAs performing basic

Reviewer: ```suggestion
- Develop adversarial experience with CHERI-enabled ISAs performing basic
```

## Skills development exercises

**Skills development exercises** are intended to take 1-2 hours each,
and ask you to build and perform minor modifications to simple C/C++
programs. These exercises facilitate building skills such as

> compiling, executing, and debugging programs with both CHERI enabled

Reviewer: ```suggestion
compiling, executing, and debugging programs with both CHERI-enabled
```

> and default ISAs, as well as to build basic understanding of CHERI

Reviewer: Hm, we should probably be consistent about whether they're "default", "baseline", "unmodified" or "non-CHERI-enabled" ISAs (all used here)

Author: I could do that as a separate editing pass over the whole book.

Author: I believe this concern is now addressed.

C/C++ properties. We highlight some key edge cases in CHERI, including
the effects of bounds imprecision, subobject bounds, weaker temporal
safety, and C type confusion.

These exercises take for granted a strong existing understanding of:
- The C/C++ languages
- UNIX program compilation, execution, and debugging

> - ARMv8 and/or RISC ISAs as well as binary structures/reverse engineering

Reviewer: ```suggestion
- RISC ISAs (e.g. Armv8-A or RISC-V) as well as binary structures/reverse engineering
```
Arm is a RISC ISA (or at least claims to be; more of a load/store architecture these days)

## Focused adversarial missions
**Focused adversarial missions** are intended to take
1-3 days, and ask you to exploit, first on an unmodified ISA, and
then on a CHERI-enabled ISA, documented vulnerabilities in simple "potted"
C/C++-language programs provided by the CHERI team. These missions
engage you more specifically in exploitation, and CHERI's
security objectives and mechanisms.

**These take for granted good existing experience with
memory-safety-related attack techniques, such as buffer overflows,
integer-pointer type confusion, Return-Oriented Programming (ROP), and
Jump-Oriented Programming (JOP).**

> Successful exploitation of non CHERI enabled variants depends only

Reviewer: ```suggestion
Successful exploitation of non-CHERI-enabled variants depends only
```
(but also see above)

upon widely published understanding and techniques (e.g., buffer
overflows combined with ROP). For those familiar with conventional
low-level attack techniques, this will also act as a warm-up exercise
on the baseline architecture and expand experience with reverse
engineering and exploitation.

> The CHERI team has confirmed exploitability for the non CHERI enabled

Reviewer: ```suggestion
The CHERI team has confirmed exploitability for the non-CHERI-enabled
```
if using that phrasing

> binaries in advance.  We strongly recommend exploiting the non CHERI
> enabled version of the code first, as a starting point for

Reviewer: ```suggestion
binaries in advance.  We strongly recommend exploiting the non-CHERI-enabled
version of the code first, as a starting point for
```
ditto (but presumably with line wrapping if using that phrasing)
